example-plugins/charting_api_examples/routes/notes.py

```diff
@@ -0,0 +1,178 @@
+                            "code": note.note_type_version.code,
+                            "system": note.note_type_version.system,
+                        },
+                    },
+                } for note in notes]
+            })
+        ]
+
+    """
+    POST /plugin-io/api/charting_api_examples/notes/
+    Headers: "Authorization <your value for 'simpleapi-api-key'>"
+    Body: {
+		"note_type_id": "c5df4f03-58e4-442b-ad6c-0d3dadc6b726",
+        "datetime_of_service": "2025-02-21 23:31:42",
+		"patient_id": "5350cd20de8a470aa570a852859ac87e",
+		"practice_location_id": "306b19f0-231a-4cd4-ad2d-a55c885fd9f8",
+		"provider_id": "6b33e69474234f299a56d480b03476d3",
+		"title": "My Note Title",
+    }
+    """
+    @api.post("/")
+    def create(self) -> list[Response | Effect]:
+        required_attributes = {
+            "note_type_id",
+            "datetime_of_service",
+            "patient_id",
+            "practice_location_id",
+            "provider_id",
+            "title",
+        }
+        request_body = self.request.json()
+        missing_attributes = required_attributes - request_body.keys()
+        if len(missing_attributes) > 0:
+            return [
+                JSONResponse(
+                    {"error": f"Missing required attribute(s): {', '.join(missing_attributes)}"},
+                    # Normally you should use a constant, but this status
+                    # code's constant changes in 3.13 from
+                    # UNPROCESSABLE_ENTITY to UNPROCESSABLE_CONTENT. Using the
+                    # number directly here avoids that future breakage.
+                    status_code=422,
+                )
+            ]
+
+        note_type_id = request_body["note_type_id"]
+        datetime_of_service = arrow.get(request_body["datetime_of_service"]).datetime
+        patient_id = request_body["patient_id"]
+        practice_location_id = request_body["practice_location_id"]
+        provider_id = request_body["provider_id"]
+        title = request_body["title"]
+
+        note_effect = NoteEffect(
+            note_type_id=note_type_id,
+            datetime_of_service=datetime_of_service,
+            patient_id=patient_id,
+            practice_location_id=practice_location_id,
+            provider_id=provider_id,
+            title=title,
+        )
+
+        return [
+            note_effect.create(),
+            JSONResponse({"message": "Note data accepted for creation"}, status_code=HTTPStatus.ACCEPTED)
+        ]
+
+    """
+    GET /plugin-io/api/charting_api_examples/notes/<note-id>/
+    Headers: "Authorization <your value for 'simpleapi-api-key'>"
+    """
+    @api.get("/<id>/")
+    def read(self) -> list[Response | Effect]:
+        note = get_note_from_path_params(self.request.path_params)
+        if not note:
+            return note_not_found_response()
+
+        status_code = HTTPStatus.OK
+        current_note_state = CurrentNoteStateEvent.objects.get(note=note).state
```

Comment:
This is included in the read and not the search because it would be difficult to avoid an n+1 on the search.